lib/bolt/transport/winrm/connection.rb

```diff
@@ -140,12 +142,59 @@ def mkdirs(dirs)
         end
 
         def write_remote_file(source, destination)
+          if target.options['file-protocol'] == 'smb'
```

Comment:
The `ruby_smb` gem only supports SMBv1/2, but not v3 (yet). However, only SMBv3 supports encryption. What should the behavior be when using `winrm+ssl`? I assume we should error here?

Reply:
More background on SMB2 vs SMB3 - https://blogs.msdn.microsoft.com/openspecification/2017/05/26/smb-2-and-smb-3-security-in-windows-10-the-anatomy-of-signing-and-cryptographic-keys/

Reply:
Agree that user asks for encryption with `+ssl` that file transfers should also be considered encrypted. So I think we have to fallback to using the winrm channel for file transfers (perhaps with a warning?) under such circumstances.

https://docs.microsoft.com/en-us/windows-server/storage/file-server/smb-security is also a bit easier to understand than the detail presented in the above links.